subprojects/build-init/src/test/groovy/org/gradle/buildinit/plugins/internal/maven/MavenProjectsCreatorSpec.groovy

```diff
@@ -141,4 +141,60 @@ class MavenProjectsCreatorSpec extends Specification {
         then:
         gradleProject.contains("compileOnly group: 'org.gradle', name: 'build-init', version:'1.0.0'")
     }
+
+    def "creates multi module project with same artifactId"() {
+        given:
+        def parentPom = temp.file("pom.xml")
+        parentPom.text = """\
+<project>
+  <modelVersion>4.0.0</modelVersion>
+  <packaging>pom</packaging>
+  <groupId>org.gradle</groupId>
+  <artifactId>test</artifactId>
+  <version>0</version>
+
+  <modules>
+    <module>commons</module>
+  </modules>
+</project>
+"""
+
+        temp.file("commons/pom.xml").text = """\
+<project>
+  <parent>
+    <groupId>org.gradle</groupId>
+    <artifactId>test</artifactId>
+    <version>0</version>
+  </parent>
+
+  <modelVersion>4.0.0</modelVersion>
+  <artifactId>commons</artifactId>
+  <packaging>pom</packaging>
+
```

Comment:
If I understand you correctly, this pom does have `project.groupId==null`

Reply:
Oh, I see. You are right. Preferably I'd love to see two different test cases or a `where` clause instead so that it's easier to see that both cases are covered.